pkg/provisioner/ironic/ironic.go

```diff
@@ -1152,6 +1152,47 @@ func (p *ironicProvisioner) GetFirmwareSettings(includeSchema bool) (settings me
 	return settings, schema, nil
 }
 
+// GetFirmwareComponents gets all available firmware components for a node and return a list.
+func (p *ironicProvisioner) GetFirmwareComponents() ([]metal3api.FirmwareComponentStatus, error) {
+	ironicNode, err := p.getNode()
+	if err != nil {
+		return nil, fmt.Errorf("could not get node to retrieve firmware components: %w", err)
+	}
+
+	// Get the components from Ironic via Gophercloud
+	var componentList []nodes.FirmwareComponent
+	var componentListErr error
+	if p.availableFeatures.HasFirmwareUpdates() {
+		componentList, componentListErr = nodes.ListFirmware(p.ctx, p.client, ironicNode.UUID).Extract()
+	} else {
+		return nil, fmt.Errorf("current ironic version does not support firmware updates")
+	}
+
+	if componentListErr != nil {
+		return nil, fmt.Errorf("could not get firmware components for node %s: %w", ironicNode.UUID, componentListErr)
+	}
+
+	// Setting to 2 since we only support bmc and bios
+	componentsInfo := make([]metal3api.FirmwareComponentStatus, 0, 2)
+
+	// Iterate over the list of components to extract their information and update the list.
+	for _, fwc := range componentList {
+		component := metal3api.FirmwareComponentStatus{
```

Comment:
If we say we only support 'bmc' and 'bios', you have to enforce this here, dropping unexpected components with a log message. Otherwise, you may product invalid HFC's.

Reply:
Since in this part  we are iterating over the return from ironic I didn't think it would be necessary

Reply:
One day, Ironic may start to return more values (this is quite likely if we find a way to e.g. update NICs). That change should not break BMO immediately.

Reply:
got it! Thanks!